t/io_uring.c

```diff
@@ -799,15 +799,15 @@ static int submitter_init(struct submitter *s)
 	int i, nr_batch, err;
 	static int init_printed;
 	char buf[80];
-
+	pthread_t tid = pthread_self();
 	s->tid = gettid();
 	printf("submitter=%d, tid=%d, file=%s, node=%d\n", s->index, s->tid,
 							s->filename, s->numa_node);
 
 	set_affinity(s);
 
-	__init_rand64(&s->rand_state, pthread_self());
-	srand48(pthread_self());
+	__init_rand64(&s->rand_state, *((unsigned long long int*) &tid));
```

Comment:
How does this work if sizeof(pthread_t) != sizeof(unsigned long long)?

Reply:
right, it seems to be ok when pthread_t size is smaller than uint64_t, but it will have aliasing issues due to the way its subverting type system :(